test/DeltaTests.cpp

```diff
@@ -112,17 +116,24 @@ std::vector<char> DeltaTests::applyDelta(
         }
         case 'd': // new run
         {
-            int destRow = (uint8_t)(delta[i+1]);
-            int destCol = (uint8_t)(delta[i+2]);
-            size_t length = (uint8_t)(delta[i+3]);
+            size_t destRow = static_cast<uint8_t>(delta[i+1]);
+            size_t destCol = static_cast<uint8_t>(delta[i+2]);
+            size_t length = static_cast<uint8_t>(delta[i+3]);
             i += 4;
 
 //            std::cout << "new " << length << " at " << destCol << ", " << destRow << '\n';
             LOK_ASSERT(length <= width - destCol);
 
             char *dest = &output[width * destRow * 4 + destCol * 4];
-            for (size_t j = 0; j < length * 4 && i < delta.size(); ++j)
-                *dest++ = delta[i++];
+            for (size_t j = 0; j < length * 4 && i < delta.size(); j += 4)
+            {
+                // Swap BGR to RGB
+                *dest++ = delta[i + 2];
```

Comment:
My expectation is that we treat a pixel as a pixel independent of its' BGRA/RGBA-ness - it's certainly far faster to handle 32bit integers in one bite.

Reply:
I think what is happening is as the png is being loaded into a byte array is loaded as BGRA. Without that change the test fails because the graphic is the wrong color (instead of light blue it becomes orange, the R and B values are swapped)

Reply:
I'll look into a better way to do this

Reply:
Ah - fair cop - if the PNG is not what we expect then we should swap that first I guess before applying a delta; but I'm surprised; possibly we're used to BGRA or somethign internally and we didn't notice yet; I'd expect PNGs to be RGB(A) though. Thanks for chasing =)